src/core/components/ChatHistory.tsx

```diff
@@ -38,19 +38,26 @@ export const ChatHistory = ({
 
   const deleteConversation = useCallback(
     (id: string) => {
-      const allConversations = JSON.parse(
-        localStorage.getItem('conversations') ?? '{}',
-      ) as Record<string, ConversationHistory>;
-
-      if (
-        allConversations[id] &&
-        id === messageHistoryStore.getConversationID()
-      ) {
+      if (id === messageHistoryStore.getConversationID()) {
         startNewChat();
       }
 
-      delete allConversations[id];
-      localStorage.setItem('conversations', JSON.stringify(allConversations));
```

Comment:
1. Update the UI with `startNewChat()`
2. Schedule all of the storage updates to happen after